saleor/checkout/checkout_cleaner.py

```diff
@@ -1,27 +1,28 @@
-from typing import Iterable, Union
+from typing import Iterable, Optional, Type, Union
 
 from django.core.exceptions import ValidationError
 
-from ...checkout.error_codes import CheckoutErrorCode
-from ...checkout.models import Checkout, CheckoutLine
-from ...checkout.utils import is_fully_paid, is_valid_shipping_method
-from ...discount import DiscountInfo
-from ...payment.error_codes import PaymentErrorCode
+from ..discount import DiscountInfo
+from ..payment import gateway, models as payment_models
+from ..payment.error_codes import PaymentErrorCode
+from .error_codes import CheckoutErrorCode
+from .models import Checkout, CheckoutLine
+from .utils import is_fully_paid, is_valid_shipping_method
 
 
 def clean_checkout_shipping(
     checkout: Checkout,
     lines: Iterable[CheckoutLine],
     discounts: Iterable[DiscountInfo],
-    error_code: Union[CheckoutErrorCode, PaymentErrorCode],
+    error_code: Union[Type[CheckoutErrorCode], Type[PaymentErrorCode]],
 ):
     if checkout.is_shipping_required():
         if not checkout.shipping_method:
             raise ValidationError(
                 {
                     "shipping_method": ValidationError(
                         "Shipping method is not set",
-                        code=error_code.SHIPPING_METHOD_NOT_SET,
+                        code=error_code.SHIPPING_METHOD_NOT_SET.value,
```

Comment:
Why are those changes needed?

Reply:
I am getting: 
```
Unexpected type(s): (str, Union[CheckoutErrorCode, PaymentErrorCode]) Possible types: (Any, Optional[str]) (Any, Optional[str]) 
```
And then it raises the exception on mypy:
```
saleor/checkout/checkout_cleaner.py:43: error: Argument "code" to "ValidationError" has incompatible type "Union[CheckoutErrorCode, PaymentErrorCode]"; expected "Optional[str]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```